notebooks/Manage Map Areas with Group and Index.ipynb

```diff
@@ -0,0 +1,456 @@
+    "if not map_areas_hfl:\n",
+    "    print(\"'Manage Map Areas' layer/table could not be found, please check MAP_AREAS_HFL_ITEM_ID...\")\n",
+    "    \n",
+    "# get group that has all the shared map area webmaps\n",
+    "manage_map_areas_group = gis.groups.get(MAP_AREAS_GROUP_ITEM_ID)\n",
+    "if not manage_map_areas_group:\n",
+    "    print(\"'Manage Map Areas' group could not be found, please check MAP_AREAS_GROUP_ITEM_ID...\")     "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Helper functions "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# send report using group notify\n",
+    "def email_group(group_id,subject,message):\n",
+    "    \n",
+    "    try:\n",
+    "        # get group based on passed in id\n",
+    "        notify_group = gis.groups.get(group_id)\n",
+    "\n",
+    "        # get group owner and add to list\n",
+    "        notification_list = [notify_group.owner]\n",
+    "        \n",
+    "        # get group members and add them to notification list\n",
+    "        for member in notify_group.get_members()[\"users\"]:\n",
+    "            notification_list.append(member)\n",
+    "\n",
+    "        # use group .notify method to send email - message supports html\n",
+    "        notify_result = notify_group.notify(users=notification_list, subject=subject, message=message,method=\"email\")\n",
+    "        \n",
+    "        return notify_result[\"success\"]\n",
+    "        \n",
+    "    except:\n",
+    "        return False"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_email_report(map_areas_hfl_item_id, processing_start_time):\n",
+    "    \n",
+    "    try:\n",
+    "        # create html table that includes the query results from Manage Map Area Processing table\n",
+    "        email_subject = \"\"\n",
+    "        email_body = \"\"\"<table border='1'>\n",
+    "                            <tr>\n",
+    "                                <th>web map</th>\n",
+    "                                <th>map area name</th>\n",
+    "                                <th>processing status</th>\n",
+    "                                <th>processing seconds</th>\n",
+    "                                <th>total storage mb</th>\n",
+    "                            </tr>\"\"\"\n",
+    "        out_fields = \"web_map_name, map_area_name,process_status,process_date,process_seconds,total_storage_mb\"\n",
+    "\n",
+    "        # get logging table\n",
+    "        map_areas_hfl = gis.content.get(map_areas_hfl_item_id)\n",
+    "        logging_table = map_areas_hfl.tables[0]\n",
+    "    \n",
+    "        # email subject - success count \n",
+    "        where_clause = f\"process_status = 'success' and process_date > '{processing_start_time}'\"\n",
```

Comment:
We could rename the where clause variables or move some things around? 

```
email_subject = f"{success_results} map areas created" if success_results else ""
if failed_results:
    if email_subject:
        email_subject += f" / {failed_results} map areas failed"
    else:
        email_subject += f"{failed_results} map areas failed"

```

Where success and failed results are your queries just without the text? Not sure if this makes it more confusing or not.  Curious what you think!

Reply:
> Where success and failed results are your queries just without the text? Not sure if this makes it more confusing or not. Curious what you think!

I did some reorganizing here - let me know what you think

> Does the user need to input min or max scale when they create a new polygon area in the HFL?

Yes - I added some text for this in setup